commands/deploy.go

```diff
@@ -289,8 +289,9 @@ Error: %s`, fprocessErr.Error())
 		}
 	} else {
 		if len(image) == 0 || len(functionName) == 0 {
-			return fmt.Errorf("To deploy a function give --yaml/-f or a --image flag")
+			return fmt.Errorf("To deploy a function give --yaml/-f or a --image and --name flag")
 		}
+		gateway = getGatewayURL(gateway, defaultGateway, "", os.Getenv(openFaaSURLEnvironment))
```

Comment:
The current code passed the `gateway` variable to the 3rd parameter but it doesn't make much sense because there isn't a `yamlURL` to be passed when doing `faas-cli deploy` with `--image` and `--name`. Hence changed it to `""`.